core/runtime/src/main/java/io/quarkus/runtime/configuration/ConfigUtils.java

```diff
@@ -20,18 +30,161 @@ private ConfigUtils() {
+        final String rawValue = config.getValue(configName, String.class);
+        return converter.convert(rawValue);
+    }
+
+    /**
+     * Retrieve the Optional value of a property represented by the given config name. Converter the value to an appropriate
+     * type using the given converter.
+     *
+     * @param config - Configuration object (must not be {@code null})
+     * @param configName - the property name (must not be {@code null})
+     * @param objectType - the type of the object (must not be {@code null})
+     * @param converterClass - The converter class to use
+     * @return Optional value of appropriate type
+     */
+    public static <T> Optional<T> getOptionalValue(SmallRyeConfig config, String configName, Class<T> objectType,
+            Class<Converter<T>> converterClass) {
+        if (converterClass == null) {
+            return config.getOptionalValue(configName, objectType);
+        }
+
+        final Converter<T> converter = getConverterOfType(objectType, (Class) converterClass);
+        final String rawValue = config.getValue(configName, String.class);
+        return Optional.ofNullable(converter.convert(rawValue));
+    }
+
+    /**
+     * Retrieve the value of a given config name from Configuration object. Converter the value to an appropriate type using the
+     * given converter.
+     *
+     * @param config - Configuration object (must not be {@code null})
+     * @param configName - the property name (must not be {@code null})
+     * @param objectType - the type of the object (must not be {@code null})
+     * @param converterClass - The converter class to use
+     * @return the values in appropriate type
+     */
+    public static ArrayList<?> getValues(SmallRyeConfig config, String configName, Class<?> objectType,
+            Class<Converter<?>> converterClass) {
+        if (converterClass == null) {
+            return config.getValues(configName, objectType, ArrayListFactory.getInstance());
+        }
+
+        final Converter<?> converter = getConverterOfType(objectType, converterClass);
+        final ArrayList<String> rawValues = config.getValues(configName, String.class, ArrayListFactory.getInstance());
+        return rawValues.parallelStream().map(converter::convert).collect(Collectors.toCollection(ArrayList::new));
+    }
+
+    /**
+     * Converter the value to an appropriate type using the given converter.
+     *
+     * @param config - Configuration object (must not be {@code null})
+     * @param value - the value (must not be {@code null})
+     * @param objectType - the type of the object (must not be {@code null})
+     * @param converterClass - The converter class to use
+     * @return the value
+     */
+    public static <T> T convert(SmallRyeConfig config, String value, Class<?> objectType,
+            Class<Converter<T>> converterClass) {
+        if (converterClass == null) {
+            return (T) config.convert(value, objectType);
```

Comment:
This should just be `config.convert(value, objectType)` once `objectType` is fixed.